packages/salesforcedx-vscode-core/src/commands/forceSourceDeploySourcePath.ts

```diff
@@ -114,28 +119,53 @@ export class LibraryDeploySourcePathExecutor extends DeployRetrieveLibraryExecut
         'Deploy (Beta)',
         'force_source_deploy_with_sourcepath_beta'
       );
-
+      channelService.showCommandWithTimestamp(`Starting ${this.executionName}`);
       if (this.sourceClient === undefined) {
         throw new Error('SourceClient is not established');
       }
-
-      this.sourceClient.tooling.deploy = this.deployWrapper(
-        this.sourceClient.tooling.deploy
-      );
-
       const projectNamespace = (await SfdxProjectConfig.getValue(
         'namespace'
       )) as string;
       const registryAccess = new RegistryAccess();
       const components = registryAccess.getComponentsFromPath(response.data);
-      const deployPromise = this.sourceClient.tooling.deploy({
-        components,
-        namespace: projectNamespace
-      });
+      let deployPromise: Promise<SourceDeployResult>;
+      if (projectNamespace) {
+        deployPromise = this.sourceClient.tooling.deploy(components, {
+          namespace: projectNamespace
+        }) as Promise<SourceDeployResult>;
+      } else {
+        deployPromise = this.sourceClient.metadata.deploy(
+          components
+        ) as Promise<SourceDeployResult>;
+      }
       const metadataCount = JSON.stringify(createComponentCount(components));
-      await deployPromise;
-
+      const result = (await vscode.window.withProgress(
+        {
+          title: this.executionName,
+          location: vscode.ProgressLocation.Notification
+        },
+        () => deployPromise
+      )) as SourceDeployResult;
+      const parser = new LibraryDeployResultParser(result);
+      const outputResult = parser.resultParser(result);
       this.logMetric({ metadataCount });
+      channelService.appendLine(outputResult);
+      channelService.showCommandWithTimestamp(`Finished ${this.executionName}`);
+      if (
+        result.status === DeployStatus.Succeeded ||
+        // @ts-ignore
+        result.status === ToolingDeployStatus.Completed
+      ) {
+        DeployRetrieveLibraryExecutor.errorCollection.clear();
+        await notificationService.showSuccessfulExecution(this.executionName);
+      } else {
+        handleDeployRetrieveLibraryDiagnostics(
+          // @ts-ignore
+          result,
+          DeployRetrieveLibraryExecutor.errorCollection
+        );
```

Comment:
are we moving away from using the deploy wrapper to run the command & contain all the output processing?

Reply:
@brpowell same question. a lot of the things added here can be moved to the wrapper code

Reply:
If i remember correctly we updated it to be more straightforward since we were having issues debugging/conditionally setting the api to use based on the namespace, but it was awhile ago so can't remember exactly. And since this command is the only one essentially using a wrapper it's not repeating any logic anyway. But it can be changed back if wanted.